scripts/conf.py

```diff
@@ -449,84 +443,88 @@ def install_gres_conf(lkp=lkp):
+            d["LinkSpeed"] = self.link_speed
+        return dict_to_conf(d)
+
+    def render_conf_lines(self) -> List[str]:
+        if self.empty():
+            return []
+
+        lines = [self.conf_line()]
+        for s in sorted(self.switches or [], key=lambda s: s.name):
+            lines.extend(s.render_conf_lines())
+        return lines
+
+    def empty(self) -> bool:
+        return not self.nodes and not self.switches
+
+
+def tpu_nodeset_switch_lines(lkp: util.Lookup) -> str:
+    root = Switch(name="nodeset_tpu-root", switches=[])
+
     for nodeset in lkp.cfg.nodeset_tpu.values():
-        tpuobj = TPU(nodeset)
-        static, dynamic = lkp.nodeset_lists(nodeset)
-        if tpuobj.vmcount == 1:
-            line = {
-                "SwitchName": nodeset.nodeset_name,
-                "Nodes": ",".join(filter(None, (static, dynamic))),
-            }
-            lines.extend([dict_to_conf(line)])
-            g_switches.append(nodeset.nodeset_name)
+        tpuobj = util.TPU(nodeset)
+        nodelists = filter(None, lkp.nodeset_lists(nodeset))  # (static, dynamic)
+
+        ns_switch = Switch(name=nodeset.nodeset_name)
+        if tpuobj.vmcount == 1:  # Put all nodes in one switch
+            ns_switch.nodes = ",".join(nodelists)
         else:
-            ns_switches = []
-            id = 0
-            if static:
-                for nodes in chunked(to_hostnames(static), n=tpuobj.vmcount):
-                    line = {
-                        "SwitchName": f"{nodeset.nodeset_name}-{id}",
-                        "Nodes": to_hostlist(nodes),
-                    }
-                    lines.extend([dict_to_conf(line)])
-                    ns_switches.append(f"{nodeset.nodeset_name}-{id}")
-                    id += 1
-                lines.extend([dict_to_conf(line)])
-            if dynamic:
-                for nodes in chunked(to_hostnames(dynamic), n=tpuobj.vmcount):
-                    line = {
-                        "SwitchName": f"{nodeset.nodeset_name}-{id}",
-                        "Nodes": to_hostlist(nodes),
-                    }
-                    lines.extend([dict_to_conf(line)])
-                    ns_switches.append(f"{nodeset.nodeset_name}-{id}")
-                    id += 1
-            if id > 0:
-                line = {
-                    "SwitchName": nodeset.nodeset_name,
-                    "Switches": to_hostlist(ns_switches),
-                }
-                lines.extend([dict_to_conf(line)])
-                g_switches.append(nodeset.nodeset_name)
-    if g_switches:
-        line = {
-            "SwitchName": "nodeset_tpu-root",
-            "Switches": to_hostlist(g_switches),
-        }
-        lines.extend([dict_to_conf(line)])
-    return "\n".join(filter(None, lines))
+            # Chunk nodes into sub-switches of size `vmcount`
+            ns_switch.switches = []
+            for nodelist in nodelists:
+                nodenames = util.to_hostnames(nodelist)
```

Comment:
JFMU, previously it was appending switches for static and dynamic separately. is there a reason to why we are doing this way ?

Reply:
> previously it was appending switches for static and dynamic separately.

Yes they were added separately, they still added separately

`for nodelist in nodelists:`, `nodelists` being `filter(None, (static, dynamic))`